win32ss/user/ntuser/sysparams.c

```diff
@@ -797,6 +797,86 @@ SpiNotifyNCMetricsChanged(VOID)
     return TRUE;
 }
 
+static void
+SpiLogFontAnsiToWide(const LOGFONTA *pLFA, LPLOGFONTW pLFW)
+{
+    RtlCopyMemory(pLFW, pLFA, offsetof(LOGFONTA, lfFaceName));
+    RtlMultiByteToUnicodeN(pLFW->lfFaceName, sizeof(pLFW->lfFaceName), NULL,
+                           pLFA->lfFaceName, sizeof(pLFA->lfFaceName));
+}
+
+static void
+SpiLogFontWideToAnsi(const LOGFONTW *pLFW, LPLOGFONTA pLFA)
+{
+    RtlCopyMemory(pLFA, pLFW, offsetof(LOGFONTA, lfFaceName));
+    RtlUnicodeToMultiByteN(pLFA->lfFaceName, sizeof(pLFA->lfFaceName), NULL,
+                           pLFW->lfFaceName, sizeof(pLFW->lfFaceName));
+}
+
+static UINT_PTR
+SpiNonClientMetricsAnsiToWide(const NONCLIENTMETRICSA *pNCMA, LPNONCLIENTMETRICSW pNCMW)
```

Comment:
The necessary ANSI/WIDE conversions are already done on client-side.
The kernel win32k (i.e. the function `NtUserSystemParametersInfo`) only deals with the UNICODE version of the NONCLIENTMETRICSW.